### PR TITLE
Issue/10784 blaze analytics part iii

### LINKED ISSUE #10784: Blaze Analytics
Add al Blaze related new events following the tracking plan: pe5sF9-2mb-p2

```[tasklist]
### Tasks
- [x] Add existing events from Blaze i2 to the new native screens: https://github.com/woocommerce/woocommerce-android/pull/10821
- [x] Add new events for the native flow: https://github.com/woocommerce/woocommerce-android/pull/10820
- [x] Add campaign creation success/failure tracking: #10926
```

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #10784

### Description

Final PR for Blaze Analytics. 
- Adds the missing events to handle Blaze campaign creation results. These are the main changes that should be reviewed https://github.com/woocommerce/woocommerce-android/pull/10926/commits/8c9f50d959c3cb474dcb5699381c2c0d4731be1e
- Adds back missing changes accidentally removed here: https://github.com/woocommerce/woocommerce-android/pull/10820/commits/ae11eefbff6a0b4734b59d2f1c2d6e309146114f#diff-9ca6a599d883aea20827d1df860489c7b8268aa2081c970308475898c27f8fa3 The changes related to reverting those accidental changes where already reviewed. So you can skip these commit changes: https://github.com/woocommerce/woocommerce-android/pull/10926/commits/73d47ba3a070f0a7cf19f244b58f04743226d414

### Testing instructions

1. Trigger Blaze campaign creation flow
2. Click on "Confirm details"
3. Click on "Submit Campaign"
4. Upon success, check the following is tracked: `🔵 Tracked: blaze_campaign_creation_success`

Repeat the steps, but before step 3, turn off the internet and check for the following event:
`blaze_campaign_creation_failed, Properties: {"blaze_creation_error":null` For the case of no connectivity there's no error message available so the value will be null. But we are mostly interested on API errors so this should not be an issue.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryViewModel.kt [hunk 1]
```diff
@@ -101,6 +104,7 @@ class BlazeCampaignPaymentSummaryViewModel @Inject constructor(
     }
 
     fun onSubmitCampaign() {
+        analyticsTrackerWrapper.track(stat = BLAZE_CREATION_PAYMENT_SUBMIT_CAMPAIGN_TAPPED)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryViewModel.kt]: I would just move this below the `if` block, WDYT?

**Review by hichamboushaba — APPROVED** [review]
Nice work, thanks @JorgeMucientes, I left a small suggestion, but it's not blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryViewModel.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryViewModel.kt]: Yeah makes sense! Applied 👍🏼

**Pull request merged**